bin/reth/src/node/mod.rs

```diff
@@ -160,19 +170,102 @@ impl Command {
             });
 
         if let Some(tip) = self.tip {
-            debug!("Tip manually set: {}", tip);
+            debug!(target: "reth::cli", %tip, "Tip manually set");
             consensus.notify_fork_choice_state(ForkchoiceState {
                 head_block_hash: tip,
                 safe_block_hash: tip,
                 finalized_block_hash: tip,
             })?;
+        } else {
+            warn!(target: "reth::cli", "No tip specified. reth cannot communicate with consensus clients, so a tip must manually be provided for the online stages with --debug.tip <HASH>.");
         }
 
         // Run pipeline
-        info!("Starting pipeline");
+        info!(target: "reth::cli", "Starting sync pipeline");
         pipeline.run(db.clone()).await?;
 
-        info!("Finishing up");
+        info!(target: "reth::cli", "Finishing up");
         Ok(())
     }
 }
+
+/// The current high-level state of the node.
+#[derive(Default)]
+struct NodeState {
+    /// The number of connected peers.
+    connected_peers: usize,
+    /// The stage currently being executed.
+    current_stage: Option<StageId>,
+    /// The current checkpoint of the executing stage.
+    current_checkpoint: BlockNumber,
+}
+
+/// A node event.
+enum NodeEvent {
+    /// A network event.
+    Network(NetworkEvent),
+    /// A sync pipeline event.
+    Pipeline(PipelineEvent),
+}
+
+impl From<NetworkEvent> for NodeEvent {
+    fn from(evt: NetworkEvent) -> NodeEvent {
+        NodeEvent::Network(evt)
+    }
+}
+
+impl From<PipelineEvent> for NodeEvent {
+    fn from(evt: PipelineEvent) -> NodeEvent {
+        NodeEvent::Pipeline(evt)
+    }
+}
+
+/// Displays relevant information to the user from components of the node, and periodically
+/// displays the high-level status of the node.
+async fn handle_events(mut events: impl Stream<Item = NodeEvent> + Unpin) {
+    let mut state = NodeState::default();
+
+    let mut interval = tokio::time::interval(Duration::from_secs(30));
+    interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
+    loop {
+        select! {
+            Some(event) = events.next() => {
+                match event {
+                    NodeEvent::Network(NetworkEvent::SessionEstablished { peer_id, status, .. }) => {
+                        state.connected_peers += 1;
+                        info!(target: "reth::cli", connected_peers = state.connected_peers, peer_id = %peer_id, best_block = %status.blockhash, "Peer connected");
+                    },
+                    NodeEvent::Network(NetworkEvent::SessionClosed { peer_id, reason }) => {
+                        state.connected_peers -= 1;
+                        let reason = reason.map(|s| s.to_string()).unwrap_or("None".to_string());
```

Comment:
nit: you should be able to get rid of the `None.to_string` by doing this in two steps

`let reason = reason.map(|s| s.to_string())`;
let reason = `reason.as_deref().unwrap_or("None")`